sonar-ballerina-plugin/src/main/resources/io/ballerina/sonar/ballerina-4.html

```diff
@@ -0,0 +1,36 @@
+<p>
+    Class methods can be isolated. An isolated method is same as an isolated function with self treated as a parameter.
+    A call to a isolated method is concurrency-safe if it is called with arguments that are safe at least until the call returns.
+    A public method needs to be isolated because it might have been intended to be used in a concurrent environment.
+</p>
+
+<h2>Noncompliant Code Example</h2>
+
+<pre>
+class EvenNumber {
+    int i = 1;
+
+    public function generate() returns int {
+        lock {
+            return self.i * 2;
+        }
+    }
+}
+</pre>
+
+<h2>Compliant Solution</h2>
+
+<p>
+    Mark public non-isolated methods as isolated if they are not the entry point of the program to ensure thread safety.
```

Comment:
"not the entry point" is not entirely accurate - e.g., for service methods

Reply:
Removed that part with f4243e9c45fa8560e64f962cde4abe6c4cf69e01